pyuvm/s20_pyuvm_reg.py

```diff
@@ -0,0 +1,219 @@
+# Main Packages same as import uvm_pkg or uvm_defines.svh
+from pyuvm import uvm_object
+from pyuvm.s18_pyuvm_reg_block import *
+from pyuvm.s21_pyuvm_reg_map import pyuvm_reg_map
+from pyuvm.s24_pyuvm_reg_includes import *
+
+# Class declaration
+class pyuvm_reg(uvm_object):
+    ## Constructor
+    def __init__(self, name="pyuvm_reg", reg_width=32):
+        super().__init__(name)
+        self._parent    = None
+        self._fields    = []
+        self._err_list  = []
+        self._mirrored  = 0
+        self._desired   = 0
+        self._reset     = 0
+        self._sum       = 0
+        self._name      = name
+        self._header    = name + " -- "
+        self._address   = "0x0"
+        self._path      = ""
+        self._width     = reg_width
+        ## If set those 2 flags will override fields values, and if set to True the fields will report error reponse
+        ## in case of (Operation,Access) expect an error
+        self.throw_error_on_read    = False;
+        self.throw_error_on_write   = False;       
+        ## Internal varibales used to detect if an operation is in progress -- thwere will be no difference between read and write
+        ## we cannot read and write from to the same register at tsame time
+        self._op_in_progress = False
+
+    # configure
+    def configure(self, parent: pyuvm_reg_block, address, hdl_path, throw_error_on_read=False, throw_error_on_write=False):
+        self._parent    = parent
+        self._address   = address
+        self._path      = hdl_path
+        self._sum       = 0
+        ## If set those 2 flags will override fields values, and if set to True the fields will report error reponse
+        ## in case of (Operation,Access) expect an error        
+        self.throw_error_on_read    = throw_error_on_read;
+        self.throw_error_on_write   = throw_error_on_write;
```

Comment:
You do not see it using because of no test at moment performed for the read write. This will emulate the concept of send an error that usually register interfaces have. So the idea behind it is using those to control the expected error

Reply:
ok. makes sense.